docs: Clarify setup() and teardown() usecases

vdaysky opened the pull request:
# Thanks for contributing to our Studio Documentation!

To ensure quality and consistency, please complete the checklist below. Provide links and details where requested. This helps us verify everything is in order.

Please ignore any options that aren't applicable to your change.

## Documentation 

- [x] I have checked grammar and flow to be sure it's simple to understand
- [x] I have verified that all terms are consistent.
- [ ] I have confirmed all links are accurate and up to date.
- [ ] I have ensured each section includes all necessary details and meets the outlined objectives.
- [ ] I have checked that any images, diagrams, or tables are clear, correctly formatted, and enhance understanding.
 
## Testing

- [x] Required - I have ran this locally and it works.
- [ ] I have verified functionality across all supported devices and browsers.
- [ ] I have verified that existing features remain unaffected by recent changes.

Once completed, share all Pull Request (PR) links in the #documentation-chat channel for review by other partners and the Mineplex development team. Thank you for your effort in improving our documentation!

vdaysky committed:
docs: Clarify setup() and teardown() usecases

BillyDotWS commented on contents/docs/sdk/custom/index.mdx:
```diff
@@ -52,4 +52,10 @@ Creating your own custom `MineplexModule`s is almost effortless! Simply create a
+<Note type="note">
+    Registering a module using `MineplexModuleManager` will call it's `setup()` method.
+    If module is a `org.bukkit.event.Listener`, it will be registered with Bukkit as well.
+    Similarly, destroying a module will call its `teardown()` method, also unregistering events.
```
'destroying a module'?

BillyDotWS commented on contents/docs/sdk/features/game/mechanics/index.mdx:
````diff
@@ -36,4 +36,9 @@ public class ExampleGame implements MineplexGame {
+```
+
+<Note type="warning">
+    Constructing a `GameMechanic` through `MineplexGameMechanicFactory` will not cause Studio Engine to call neither of its `setup()` nor `teardown()` methods.
````
`neither of its` -> `either`

BillyDotWS replied:
I think it's worth improving the examples on this page too whilst you're here

BillyDotWS reviewed (changes requested):
A couple minor changes

vdaysky committed:
docs: Restructure mechanics page

vdaysky commented:
I made some changes, wanted to know your opinion on those - if you don't agree I'll just rollback. I feel like this new structure would be more intuitive, and I am saying this as someone who spent a lot of time staring at those docs, trying to understand how things work. I acknowledge that it is possible that you already considered this structure and decided against it, and if so I would be curious to know the reasoning.

I renamed built-in mechanics page into just mechanics, and moved all of the mechanics docs from game engine module page to mechanics page (potentially if you don't like the nesting mechanics page can be moved to uppermost sdk section level for better visibility, but since mechanics are coupled heavily with the game having it as nested page also makes sense. Having it inside game engine page though is too much IMO. Currently that page has information on game cycle, game interface, game states, mechanics, custom mechanics, mechanic factories, has two 175-line listings that have 4 lines of diff between them and it also documents GameWorldSelectorMechanic for some unknown reason). Also weirdly enough path of the mechanics page (/features/game/mechanics) was already implying that it is sub page of game engine module (I am starting to think that it was, but someone changed it while preserving path for backwards compatibility). 

Another side effect of moving both custom and built-in mechanics to one page is that same has to be done for modules, and honestly I think that would be a great change. There are two separate pages for built-in and custom modules, while they are really the same thing, with built-in being custom modules implemented and registered by mineplex (also both pages are relatively empty). On an unrelated note, why is that page called "Features" in navbar, while it contains modules and index page is titled as modules?

vdaysky commented on contents/docs/sdk/features/game/mechanics/index.mdx:
```diff
@@ -1,19 +1,136 @@
+Implementing your own custom `GameMechanic`s is easy to do: just create a new class that implements the `GameMechanic`
+interface, and determine if you want a mechanic that is coupled to a game, for example `BossMechanic<BossGame>`, or a
+more generic mechanic, such as `SpectatorMechanic<MineplexGame>`. In your custom class, you will need to implement
+the `setup(Game)`, `teardown()`, and `onStateChange(GameState, GameState) - todo: wtf is this` methods to define the custom behavior of your
```
What is this `onStateChange` though? Is it something from a super old sdk version?

BillyDotWS replied:
This appears to be removed, likely for handling state changes on older versions
